community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -523,7 +529,7 @@ private static void writerHeader( PagedFile pagedFile, Header.Writer headerWrite
         // Write/carry over header
         int headerOffset = cursor.getOffset();
         int headerDataOffset = headerOffset + Integer.BYTES; // will contain length of written header data (below)
-        if ( otherState.isValid() )
+        if ( otherState.isValid() || headerWriter != CARRY_OVER_PREVIOUS_HEADER )
```

Comment:
Why do you check for `CARRY_OVER_PREVIOUS_HEADER` here? Could you not use it instead?